README.md
# Von
> a personal blog for ACG lover
>>一个基于Vue开发的博客，献给所有热爱ACG，热爱生活的人；也算是给几月来坚持不懈的我的奖赏吧

**该项目为重构优化之后的SSR版本，并且以后也只维护该项目，如对旧项目（SPA版本）有需要请转[此处](https://github.com/Bersder/nameless-blog)**
## Introduction
整个博客（下称Von）前端完全基于 Vue+Nuxt 开发，后端接口层 php 实现，python 辅助数据采集

除了一般博客有的功能，Von 还有如下特点
- 技术上
  - 自适应宽度，**双端适配**（PC+移动）
  - **服务端渲染**（SSR），更好的SEO
  - ...
- 内容上
  - **追番页面**（bangumi同步），记录追番进程
  - **动态页面**（仿自B站），发送带图动态，分享日常
  - ...
  
版本更新记录 [CHANGE_LOG](./CHANGE_LOG.md)

更详细的功能介绍转[博客文章](https://oshinonya.com/archive/code/1)
## Getting Started（待完善
该博客安装分为前端和后端两部分进行
### Back-end
后端需要 LNPM 环境，宝塔之类的一键安装应该也可以吧（没试过），如果你喜欢折腾，可以参考[这篇文章](https://oshinonya.com/archive/code/2)（我的生产环境就是基于此的）

LNPM 部署并配置好后
- 执行sql脚本（见后端项目）初始化数据库
- 在网站根目录中解压该压缩包以初始化文件资源
- 下载对应版本的后端代码放在网站根目录，重命名为`apis`，并[按照要求进行配置](https://github.com/Bersder/Von-backend)

接下来安装 nodejs，很简单这里就跳过了，下面给出之后要安装的模块
```bash
npm install pm2 -g
```
pm2 是用来托管进程的，能保证你和server的远程连接断开后 nuxt 还在运行。安装完pm2后把执行文件链到 bin 里以在命令行中使用：`ln -s nodejs目录/bin/pm2 /usr/local/bin/pm2` 
___
最后是配置 Nginx 进行反向代理，把对页面的请求交给 nuxt 服务端，对静态资源和接口的请求进行拦截。向 http 块下添加以下配置，部分配置按实际情况添加 or 修改
```nginx
upstream nuxtserver{
    server 127.0.0.1:8080;
    keepalive 64;
}
server {
    listen       80;
    # 网站域名
    server_name  www.xxx.com, xxx.com;
    index  index.html index.htm;
    # 禁止文件(夹)
    location ^~ /node_modules {
        return 404;
    }
    # 静态资源
    location ~ ^/(downloads|music|site|static|tmp|uploads)/ {
        # 网站根目录，下同
        root html_nuxt;
        access_log off;
        # 静态资源允许跨域，设置缓存
        expires 30d;
        add_header Access-Control-Allow-Origin *;
        add_header Access-Control-Allow-Headers X-Requested-With;
        add_header Access-Control-Allow-Methods GET,POST,OPTIONS;
    }
    # PHP CGI
    location ~ \.php$ {
        root           html_nuxt;
        fastcgi_pass   127.0.0.1:9000;
        fastcgi_index  index.php;
        fastcgi_param  SCRIPT_FILENAME $document_root$fastcgi_script_name;
        include        fastcgi_params;
    }
    # 首先查看有没有该文件
    location / {
        root html;
        try_files $uri @nuxt;
    }
    # 最后交由代理
    location @nuxt{
        proxy_redirect off;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_read_timeout 1m;
        proxy_connect_timeout 1m;
        proxy_pass http://nuxtserver;
    }
    error_page   500 502 503 504  /50x.html;
    location = /50x.html {
        root   html_nuxt;
    }
}
```
___
另：如果你有 https 的需要可以参考申请免费证书的[教程](https://oshinonya.com/archive/code/3)
### Front-end
前端代码打包前进行自定义修改，都是些个人信息相关的东西（目前这部分信息的修改方式还未独立出来，只能改一次打包一次，未来版本会更新）
- 按需修改 `config/user.conf.js`
- 修改 `pages/about/index.vue` 的模板（关于页面）
- 修改 `pages/links/index.vue` 的模板（友链页面）
- 修改 `staitc/robots.txt` 中sitemap指向

修改完后**对整个项目除`nuxt.config.js`和`README.md`的所有文件进行正则替换，把`'/root'\+|/root`替换为空（共47个分布在27个文件中）**，到此代码部分修改完成，进入打包环节。

先`npm install`安装项目依赖，然后在项目根目录执行 `npm run build` 打包生成 `.nuxt` 文件夹。将 **.nuxt, nuxt.config.js, package.json, package-lock.json, static下的所有文件** 上传至服务器网站根目录

### Run
到此网站根目录应该如下
```text
www_root
  |--apis/
  |--apple-icon.png
  |--downloads/
  |--favicon.ico
  |--music/
  |--node_modules/
  |--.nuxt/
  |--nuxt.config.js
  |--package.json
  |--package-lock.json
  |--robots.txt
  |--site/
  |--sitemap.xml
  |--static/
  |--tmp/
  |--uploads/
```
再次检查php-fpm、nginx、mariadb/mysql服务是否已开启，如果没有问题，命令行执行`pm2 start npm --name "自定名称" -- run start`让pm2托管SSR服务，再执行`pm2 save`保存当前任务实现自启

## Support
Nothing Yet
## Author
**Von** © [Bersder](https://github.com/Bersder), Released under the **MIT** License
>[Blog](https://oshinonya.com) · GitHub [@Bersder](https://github.com/Bersder) · Twitter [@nya_oshino](https://twitter.com/nya_oshino) · QQ 1747322151


CHANGE_LOG.md
# 版本更新记录
## to do list
> 画饼一时爽，一直画饼一直爽

- [ ] 重大修改：文章分类重组（ACGN，极客，随写），修改前后端不再使用string来判别类型 ``
- [ ] Bug：修复appendChild报错（好像现在没事了？
- [ ] 弹幕实现
- [ ] 前端速度/性能优化，长列表优化，seo深入研究

## v1.2.5
`2020-`_``
### Features
- [X] sitemap 支持与自动更新 `f7b562f`
- [X] UI 优化 `0844a4f、39e953e、854b59f、f1773aa`
- [X] 二次评论自动记住账号 `ad69416`
- [ ] 新增 文章/笔记 隐藏功能
## Bug Fixes
- [X] 修复SSR端404错误 `ac5f91a`

## v1.2.4
`2020-07-31`_`3fd34da`
### Features
- [X] 文章编辑页添加离开/刷新提示 `518d9e9`
- [X] 字体更新 `518d9e9`
- [X] highlight、katex库改用国内CDN `9e07692`
## Bug Fixes
- [X] 修复Header中标题、图片样式切换不同步问题 `af8b4ea`
- [X] 修复文章编辑页取消返回时url不恢复的问题 `8e7900b`
- [X] 修复无法正确修改cookie的问题，完善cookie管理api `97b9c0a`
- [X] 修复评论预览代码块不渲染问题 `9e07692`
- [X] 修复评论区、动态区markdown样式支持不全(ol...); 友链、关于页面适用 `80b62e4、3fd34da`

## v1.2.3
`2020-05-20`_`bc643da`
### Features
- [X] 文章笔记页Header改为贴顶（导航栏贴顶透明） `a13b907`
- [X] 个人空间首页填坑：新增站点统计 `bc643da`
## Bug Fixes
- [X] 修复窄屏下刷新顶部导航栏会跳动（先出现普通导航栏，之后跳变为mini导航栏）的问题 `a3c0fc3`

## v1.2.2
`2020-04-23`_`e39f3a2`
### Features
- [X] 将等待动画组件化 `0717f0e`
- [X] 动态区移动端适配 `312785d`
- [X] 标签页面开辟子路由，避免全页刷新 `e39f3a2`

## v1.2.1
`2020-04-08`_`a62ccae`

**PRETTIFY——UI体验优化**
### Features
- [X] 实现动态侧边栏貼顶过渡效果 `5e6bfff`
- [X] ACG页面文章卡片美化 `4803d7b`
- [X] 样式优化（笔记/标签/归档页面） `a62ccae`

## v1.2.0
`2020-03-31`_`c7b6976`
### Features
- [X] 船新Nuxt重构版本，炸裂优化！

## v1.2.0之前的版本
请前往[旧项目](https://github.com/Bersder/nameless-blog/blob/master/CHANGE_LOG.md)查看


config/user.conf.js
export default {
  // TODO: 考虑将大部分移动至后端数据库
  rootDomain:'oshinonya.com',//网站根域名
  siteURL:'https://oshinonya.com',//网站地址
	sinceTime:1573659213000, //网站始于 2019.11.13 23:33:33 GTM+8
	birthday:'1999-06-05', //生日，用于计算年龄
  email:'oshinoNya@outlook.com',//联系邮箱
	statusMap:[
		{icon:'icon-coffee',des:'享受',color:'#b77d68'},
		{icon:'icon-focus',des:'不咕鸟',color:'#ff3d44'},
		{icon:'icon-gezi',des:'咕咕咕',color:'#00a1d6'},
		{icon:'icon-bxs-smiley-sad',des:'(悲)',color:'#9b9b9b'}
	],
	socialList:[//首页社交链接
		{title:'github',href:'https://github.com/Bersder',imgSrc:'/static/images/sns/github.png'},
		{title:'netease',href:'https://music.163.com/user/home?id=93044810',imgSrc:'/static/images/sns/netease.png'},
		{title:'bilibili',href:'https://space.bilibili.com/13351138',imgSrc:'/static/images/sns/bilibili.png'},
		{title:'twitter',href:'https://twitter.com/nya_oshino',imgSrc:'/static/images/sns/twitter.png'},
		{title:'email',href:'mailto:oshinoNya@outlook.com',imgSrc:'/static/images/sns/email.png'},
	],
	friendExchange:[//Game页面右侧游戏好友交换
		{icon:'iconfont icon-playstation',description:'PSN',value:'Bersder3000'},
		{icon:'iconfont icon-ns',description:'NS',value:'SW-1724-8480-2131'},
		{icon:'iconfont icon-steam',description:'steam',value:'Bersder3000'}
	],
	languageList:[//Code页面右侧语言列表
		{lang:'python',imgSrc:'/static/images/lang/moe.python.jpg',description:'python是最好的语言'},
		{lang:'c',imgSrc:'/static/images/lang/moe.c.png',description:'这是什么，我有学过吗(滑稽)'},
		{lang:'php',imgSrc:'/static/images/lang/moe.php.jpg',description:'破站后端支持'},
		{lang:'js',imgSrc:'/static/images/lang/moe.javaScript.jpg',description:'...'},
	],
	siteHistory:[//关于页面的网站黑历史
		['????-??-??','学Vue时用Vue写了一个页面，自认为还不错，加之有建博客的想法，然后一发不可收拾...'],
		['2019-11-13','经过5个月的摸爬滚打，初版博客最终完成（其实是不想再拖了，实际上想干的是还是很多）'],
		['2019-12-16','新添番组页面，与 bangumi 账户维持同步，展示追番进度'],
		['2020-02-06','新增动态页面，可以发带图动态了'],
		['2020-03-30','破站 nuxt 重构完成，唉，坑真不是一般的多'],
	],
	siteTitle:{//网站标题显示
		title:'忍野喵',
		title_:' | 忍野喵',
		subTitle:'おしの · にゃ'
	}
}
